### PR TITLE
Custom palette cairo

### PR DESCRIPTION
**Pull request by matthiasclasen**

### CODE REVIEW

File: util/helper-cairo.hh [hunk 1]
```diff
@@ -125,6 +125,18 @@ helper_cairo_create_scaled_font (const font_options_t *font_opts)
...
+#ifdef CAIRO_COLOR_PALETTE_CUSTOM
+  if (font_opts->custom_palette)
+  {
+    char **entries = g_strsplit (font_opts->custom_palette, ",", -1);
+    for (unsigned int i = 0; entries[i]; i++)
+    {
+      unsigned int idx, fr, fg, fb, fa;
+      if (sscanf (entries[i], "%u:%2x%2x%2x%2x", &idx, &fr, &fg, &fb, &fa) == 5)
+        cairo_font_options_set_custom_palette_color (font_options, idx, fr / 255., fg / 255., fb / 255., fa / 255.);
+    }
```
Comment (matthiasclasen) [inline util/helper-cairo.hh:137]: missing a g_strfreev (entries);

**Comment by behdad** [comment]
Cairo counterpart: https://gitlab.freedesktop.org/cairo/cairo/-/merge_requests/412

File: util/view-options.hh [hunk 2]
```diff
@@ -111,8 +111,8 @@ view_options_t::add_options (option_parser_t *parser)
     {"annotate",	0, 0, G_OPTION_ARG_NONE,	&this->annotate,		"Annotate output rendering",				nullptr},
     {"background",	0, 0, G_OPTION_ARG_STRING,	&this->back,			"Set background color (default: " DEFAULT_BACK ")",	"rrggbb/rrggbbaa"},
     {"foreground",	0, 0, G_OPTION_ARG_STRING,	&this->fore,			"Set foreground color (default: " DEFAULT_FORE ")",	"rrggbb/rrggbbaa"},
-    {"font-palette",    0, 0, G_OPTION_ARG_INT,         &this->palette,                 "Set font palette (default: 0)",                "index"},
-    {"custom-palette",  0, 0, G_OPTION_ARG_STRING,      &this->custom_palette,          "Custom palette",                               "comma-separated colors"},
+    {"font-palette",    0, 0, G_OPTION_ARG_INT,         &this->palette,                 "Set font palette (default: 0, custom: -1)",            "index"},
```
[the remark this answers is not in the document]
Comment (matthiasclasen) [reply util/view-options.hh]: oh, ok

File: meson.build [hunk 3]
```diff
@@ -214,6 +214,7 @@ if cairo_dep.found()
   else
     check_funcs += [['cairo_user_font_face_set_render_color_glyph_func', {'deps': cairo_dep}]]
   endif
+  check_funcs += [['cairo_font_options_get_custom_palette_color', {'deps': cairo_dep}]]
```
Comment (khaledhosny) [inline meson.build:217]: This should go inside the else branch above, as we can't use this when we are building subproject.

Comment (behdad) [reply meson.build:217]: I'll move there.  Need to roll forward the subproject dependency when the API lands in cairo.

Comment (khaledhosny) [reply meson.build]: Indeed, once it lands we can do that.

Comment (matthiasclasen) [reply meson.build]: It landed

**Pull request merged**